From ecb30548cbfe3791481845405d45e1050f56400d Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Mon, 20 Oct 2025 13:43:12 +0300
Subject: [PATCH] video: driver: stop playing tricks with Kbuild

Source management systems (e.g. quilt) can create subdirs with copies of
headers in them, which poison compiler include paths due to a generous
`find` invocation in Kbuild. As Kbuild already lists all header dirs,
drop extra tricks and set ccflags-m explicitly.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://github.com/qualcomm-linux/video-driver/pull/20]
---
 Kbuild | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/Kbuild b/Kbuild
index a2f5cd95d6ac..f3fc7be60f50 100644
--- a/Kbuild
+++ b/Kbuild
@@ -8,13 +8,8 @@ ifneq ($(KBUILD_EXTRA_CONFIGS),)
 include $(KBUILD_EXTRA_CONFIGS)
 endif
 
-# List of all camera-kernel headers
-video_include_dirs := $(shell dirname `find $(VIDEO_KERNEL_ROOT) -name '*.h'` | uniq)
-
 # Include Kernel headers
-LINUXINCLUDE +=                                 \
-    -I$(KERNEL_ROOT)                            \
-    $(addprefix -I,$(video_include_dirs))         \
+ccflags-m +=                                 \
     -I$(VIDEO_KERNEL_ROOT)/vidc/inc \
     -I$(VIDEO_KERNEL_ROOT)/variant/common/inc \
     -I$(VIDEO_KERNEL_ROOT)/variant/iris2/inc \
@@ -24,9 +19,7 @@ LINUXINCLUDE +=                                 \
     -I$(VIDEO_KERNEL_ROOT)/platform/sm8550/inc \
     -I$(VIDEO_KERNEL_ROOT)/platform/sa8775p/inc \
     -I$(VIDEO_KERNEL_ROOT)/platform/qcs8300/inc \
-    -I$(VIDEO_KERNEL_ROOT)/include/uapi/vidc/media \
     -I$(VIDEO_KERNEL_ROOT)/include/uapi/vidc/ \
-    -I$(VIDEO_KERNEL_ROOT)/
 
 # After creating lists, add content of 'ccflags-m' variable to 'ccflags-y' one.
 ccflags-y += ${ccflags-m}
-- 
2.47.3

